onnxruntime/core/optimizer/graph_output_deduplication.cc

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// graph_output_deduplication.cc
+
+#include "core/optimizer/graph_output_deduplication.h"
+#include <queue>
+
+namespace onnxruntime {
+
+/*
+  Deduplication logic for graph outputs:
+  if this node produces a graph output which is a duplicate output, introduce an identity
+  node in front of each such duplicated output keeping the graph output order consistent
+  before and after transformation.
+
+  Example:
+
+    An Add op shown below with duplicate graph outputs as X
+                  Add
```

Comment:
I saw comments in https://github.com/microsoft/onnxruntime/issues/3141 from @yufenglee regarding to similar issue:
```
To summarize the issue,
PyTorch ONNX exporter will fix the duplicated output name issue.
ONNX will add a check that there is no duplicate input and output names
ONNXRuntime will add a check and fail early if model has duplicate input or output names.
``` 
Shall we follow that path to fix such issue?

Reply:
This could potentially be addressed in the exporter by adding an identity to ensure output names are not duplicated. However, this issue is blocking a model from onboarding with ```ORTModule```. So, while the exporter team tries to prioritize and address this on the exporter side, it might be worth having this graph transformation in ORT so we can unblock users.

@garymm FYI.

Reply:
If exporter team can fix it, perhaps we can modify torch code as workaround for now?